public/js/CollisionManager.js

```diff
@@ -26,8 +26,61 @@ class CollisionManager {
 		this.ZerlinOnPlatform();
 	}
 
+	/* On droids colliding, swap deltaX and deltaY */
 	droidOnDroid() {
-
+		//check collision with other droids.
+		//can optimize by not checking every single droid with each other twice.
+        
+		// for (var i = this.game.droids.length - 1; i >= 0; i--) {
+		// 	var droid1 = this.game.droids[i];
+		// 	for (var j = this.game.droids.length - 1; j >= 0; j--) {
+		// 		var droid2 = this.game.droids[j];
+		// 		if(droid1 != null && droid2 != null && droid1 != droid2 
+		// 			&& collideCircleWithCircle(droid1.boundCircle.x, droid1.boundCircle.y, droid1.boundCircle.radius,
+		// 				droid2.boundCircle.x, droid2.boundCircle.y, droid2.boundCircle.radius)) {
+
+		// 			//handle collision of droids with droids. 
+		// 			//swap velocities, bounce effect.
+		// 			/*
+		// 			* BUG ALERT:
+		// 			* kinda strange, has the characteristics that we want as in that
```

Comment:
What's the bug/ feature? I didn't understand. Are you saying that you didn't actually program it to only collide some of the time, but is doing so anyway?